### PR TITLE
Implement Rule-00834 on Age and Age Units

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**

### CODE REVIEW

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js [hunk 1]
```diff
@@ -24,13 +25,15 @@ export default class ClientInformation extends React.Component {
       selected: [],
       age: '',
       ageUnitValue: '',
+      dateOfBirth: '',
     };
   }
 
   componentWillReceiveProps(nextProps) {
     let childClient = nextProps.childClient;
```
Comment (kelfink) [inline app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: Could skip the variable and assign state directly.

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js [hunk 2]
```diff
@@ -43,6 +46,19 @@ export default class ClientInformation extends React.Component {
     });
   };
 
+  checkDOB = event => {
+    let BirthDate = this.state.childClient.birth_dt;
```
Comment (kelfink) [inline app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: Better to name this dateOfBirth, or dateBirth.  Not the uppercase B to begin.

File: app/javascript/_components/DropDown/DropDown.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (kelfink) [inline app/javascript/_components/DropDown/DropDown.js]: Are we cloning DropDownField from react-wood-duck?  Are we not able to use the existing DropDownField component ?

Comment (RamuKammagani-osi) [reply app/javascript/_components/DropDown/DropDown.js]: React-wood-duck component doesn't have the disable prop, so i have just created the component by adding the prop, so while we implement all the rules we may have chance to add some more props. Once we get clarity on adding what props what we need, then ill make those changes in the existing component in react-wood-duck and submit PR react-wood-duck.

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js (hunk 2 above)
Comment (RamuKammagani-osi) [reply app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: will do, Thanks

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js (hunk 1 above)
Comment (RamuKammagani-osi) [reply app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: Changes Updated! Thanks

File: app/javascript/_components/DropDown/DropDown.js (hunk 3 above)
Comment (kelfink) [reply app/javascript/_components/DropDown/DropDown.js]: And then you would throw away this "temporary". dropdown?  that sounds good.

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js [hunk 4]
```diff
@@ -198,15 +218,18 @@ export default class ClientInformation extends React.Component {
...
-              onChange={this.handleOnChange('age')}
+              onChange={this.handleAge}
+              disabled={childClient.estimated_dob_code === 'N'}
             />
-            <DropDownField
-              id="dropdown4"
+
+            <DropDown
+              id="dropdown3"
+              label="Age Units"
```
Comment (kelfink) [inline app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: label change ... plural ?  That seems strange.

**Pull request merged**